metadata_layer.py

```diff
@@ -361,6 +368,46 @@ def __init__(self, content_type, drm_scheme, link=None):
             )
         self.link = link
 
+
+class RecommendationData(object):
+
+    def __init__(self, data_source, identifiers=None):
+        self.data_source = data_source
+        self.identifiers = identifiers or []
+
+    @property
+    def recommended_works(self):
+        """:return: a query of all of the recommended works in self.identifiers
+        or None
+        """
+        identifier_ids = []
+        _db = Session.object_session(self.data_source)
+        for identifier_data in self.identifiers[:]:
+            if isinstance(identifier_data, Identifier):
+                identifier = identifier_data
+            else:
+                identifier, ignore = Identifier.for_foreign_id(
+                    _db, identifier_data.type, identifier_data.identifier,
+                    autocreate=False
+                )
+            if not identifier:
+                self.identifiers.remove(identifier_data)
+                continue
+            identifier_ids.append(identifier.id)
+
+        if not identifier_ids:
+            return None
+
+        equivalent_identifier = aliased(Identifier)
+        works_q = Work.feed_query(_db)
```

Comment:
Wow, I'd forgotten about `feed_query`. I stopped using it when I got all the "get a list of Works" code into Lanes. Since this is a "list of Works" and it's going to eventually get turned into an OPDS feed, I think it makes sense as a type of `Lane`. Maybe a subclass? Then you'd be able to get rid of `feed_query` altogether.